### PR TITLE
twister: reuse test schema in common section

### PR DESCRIPTION
**Pull request by nashif**
Rather than duplicating the same schema, define it only once and reuse
for both common section and tests section.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by hakehuang — DISMISSED** [review]

File: scripts/schemas/twister/testsuite-schema.yaml [hunk 1]
```diff
@@ -8,200 +8,209 @@
...
+    "build_on_all":
+      type: bool
+      required: false
+    "depends_on":
+      type: any
+      required: false
+    "extra_args":
+      type: any
+      required: false
+    "extra_configs":
```
Comment (golowanow) [inline scripts/schemas/twister/testsuite-schema.yaml:40]: now `extra_configs` is allowed in the common section ?

File: scripts/schemas/twister/testsuite-schema.yaml [hunk 2]
```diff
@@ -8,200 +8,209 @@
...
+      type: any
+      required: false
+    "platform_allow":
+      type: any
+      required: false
+    "platform_type":
+      type: seq
+      required: false
+      sequence:
+        - type: str
```
Comment (golowanow) [inline scripts/schemas/twister/testsuite-schema.yaml:160]: this part from the old common section is now excluded
```
- type: str
enum: ["mcu", "qemu", "sim", "unit", "native"]
```

**Review by golowanow — CHANGES_REQUESTED** [review]
I've noticed two missing parts now in regard of the old `common` section vs. the new `scenario-schema` - looks like the `tests` section was taken to reuse it there, whereas it was slightly different with the old `common` section.

Please clarify is this change intentional ?

File: scripts/schemas/twister/testsuite-schema.yaml (hunk 1 above)
Comment (nashif) [reply scripts/schemas/twister/testsuite-schema.yaml:40]: yes

**Comment by nashif** [comment]
> I've noticed two missing parts now in regard of the old `common` section vs. the new `scenario-schema` - looks like the `tests` section was taken to reuse it there, whereas it was slightly different with the old `common` section.
> 
> Please clarify is this change intentional ?

this is the main reason for this PR, there are descripencies between the sections and it is easy to introduce inconsistencies. Thanks for spotting the difference.

**Review by golowanow — APPROVED** [review]

**Review by gchwier — APPROVED** [review]

**Pull request merged**